source/common/config/subscription_factory_impl.cc

```diff
@@ -51,6 +41,24 @@ SubscriptionPtr SubscriptionFactoryImpl::subscriptionFromConfigSource(
     const envoy::config::core::v3::ApiConfigSource& api_config_source = config.api_config_source();
     Utility::checkApiConfigSourceSubscriptionBackingCluster(cm_.primaryClusters(),
                                                             api_config_source);
+    const auto transport_api_version = api_config_source.transport_api_version();
+    if (transport_api_version == envoy::config::core::v3::ApiVersion::AUTO ||
+        transport_api_version == envoy::config::core::v3::ApiVersion::V2) {
+      runtime_.countDeprecatedFeatureUse();
+      const std::string& warning = fmt::format(
+          "V2 (and AUTO) xDS transport protocol versions are deprecated in {}. "
+          "The v2 xDS major version is deprecated and disabled by default. Support for v2 will be "
+          "removed from Envoy at the start of Q1 2021. You may make use of v2 in Q4 2020 by "
+          "setting '--bootstrap-version 2' on the CLI for a v2 bootstrap file or by enabling "
+          "the runtime envoy.reloadable_features.enable_deprecated_v2_api flag.",
+          config.DebugString());
```

Comment:
I don't think it's going to be obvious to people how to fix this from this error message (how to enable v3, etc.). Can you make this a bit more obvious in this message and/or add a release note for this change to make it super clear what people have to do?

Reply:
I'll add a FAQ and link to that.